forge-game/src/main/java/forge/game/ability/effects/ConniveEffect.java

```diff
@@ -0,0 +1,94 @@
+import forge.game.ability.AbilityKey;
+import forge.game.ability.AbilityUtils;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.*;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.game.zone.ZoneType;
+import forge.util.Lang;
+
+import java.util.List;
+import java.util.Map;
+
+public class ConniveEffect extends SpellAbilityEffect {
+
+    /* (non-Javadoc)
+     * @see forge.game.ability.SpellAbilityEffect#getStackDescription(forge.game.spellability.SpellAbility)
+     * returns the automatically generated stack description string
+     */
+    @Override
+    protected String getStackDescription(SpellAbility sa) {
+        final StringBuilder sb = new StringBuilder();
+
+        List<Card> tgt = getTargetCards(sa);
+
+        sb.append(Lang.joinHomogenous(tgt)).append(tgt.size() > 1 ? " connive." : " connives.");
+
+        return sb.toString();
+    }
+
+    /* (non-Javadoc)
+     * @see forge.game.ability.SpellAbilityEffect#resolve(forge.game.spellability.SpellAbility)
+     */
+    @Override
+    public void resolve(SpellAbility sa) {
+        final Card host = sa.getHostCard();
+        final Player hostCon = host.getController();
+        final Game game = host.getGame();
+        final int num = AbilityUtils.calculateAmount(host, sa.getParamOrDefault("ConniveNum", "1"), sa);
+
+        GameEntityCounterTable table = new GameEntityCounterTable();
+        final CardZoneTable triggerList = new CardZoneTable();
+        Map<Player, CardCollectionView> discardedMap = Maps.newHashMap();
+        Map<AbilityKey, Object> moveParams = AbilityKey.newMap();
+        moveParams.put(AbilityKey.LastStateBattlefield, sa.getLastStateBattlefield());
+        moveParams.put(AbilityKey.LastStateGraveyard, sa.getLastStateGraveyard());
+
+        for (final Card c : getTargetCards(sa)) {
+            final Player p = c.getController();
+
+            p.drawCards(num, sa, moveParams);
+
+            CardCollectionView dPHand = p.getCardsIn(ZoneType.Hand);
+            dPHand = CardLists.filter(dPHand, CardPredicates.Presets.NON_TOKEN);
+            if (dPHand.isEmpty()) { // seems unlikely, but just to be safe
+                continue; // for loop over players
+            }
+
+            CardCollection validCards = CardLists.getValidCards(dPHand, "Card", hostCon, host, sa);
+
+            if (!p.canDiscardBy(sa, true)) {
+                continue;
+            }
+
+            int amt = Math.min(validCards.size(), num);
+            CardCollectionView toBeDiscarded = amt == 0 ? CardCollection.EMPTY :
+                    p.getController().chooseCardsToDiscardFrom(p, sa, validCards, amt, amt);
+
+            if (toBeDiscarded.size() > 1) {
+                toBeDiscarded = GameActionUtil.orderCardsByTheirOwners(game, toBeDiscarded, ZoneType.Graveyard, sa);
+            }
+
+            discardedMap.put(p, toBeDiscarded);
+            discard(sa, triggerList, true, discardedMap, moveParams);
```

Comment:
either `Map<Player, CardCollectionView> discardedMap = Maps.newHashMap();` needs to be inside the loop, or `discard(sa, triggerList, true, discardedMap, moveParams);` needs to be outside of the loop

also i currently don't know how it should apply for multiple creatures connive at the same time

Reply:
I put discard outside the loop - we can see if rules dictate otherwise